### ISSUE TITLE
[Client bug]: provide a descriptive title here

### ISSUE THREAD

**Issue by ranoichman** [microsoftgraph/msgraph-sdk-dotnet#1103]
**Describe the bug**
When performing some calls that might fail from several reasons (for example: trying to add servicePrincipal to role when the user doesn't have admin rights), instead of the call to fail quickly with a proper error as other calls, the call "hangs" for about 15 minutes(!!) until failing with a general exception: " Code: generalException Message: An error occurred sending the request. Can't access a disposed object".
Modifying the "httpProvider.OverallTimeout" doesn't help as nothing is changed (and the default 1:30 minutes value doesn't help either).

**To Reproduce**
1. Create a new user with no special privilege's 
2. Use the SDK to call  adClient.DirectoryRoles[directoryRole.ObjectId].Members.References.Request().AddAsync(directoryObject).Wait();
The call will now stuck for 15 minutes until failing with the general exception.

**Expected behavior**
If an attempt to perform an action where the user has no privileges, fail quickly with a proper error, or make the "httpProvider.OverallTimeout" work and fail the call due to timeout.

**Screenshots**

**Client version**
Provide the client version number from the NuGet package.

**Desktop (please complete the following information):**
 - OS: Win10
 - Browser Chrome
 - Version 92

**Additional context**

**Comment by andrueastman** [comment]
Hey @ranoichman,

Thanks for raising this. Are you able to provide more information on the version of the sdk you are using as well as the .Net runtime in use?

Also, is there any reason to not use the async await/await pattern as below?

```cs
await adClient.DirectoryRoles[directoryRole.ObjectId].Members.References.Request().AddAsync(directoryObject);
```

**Comment by ranoichman** [comment]
Hi,

We are using .net 4.7.2 and I'm not sure what SDK version but it's quite
old now, about 3 years or so I think...

Regarding the Async - Not sure exactly, but we are using these operations
as a pipeline so no real benefit of Async await would help us, unless this
might cause the issues I am seeing? (I highly doubt it though)

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by andrueastman** [comment]
I only mention this since `Task.Wait()` is sometimes known to cause deadlocks. See [here](https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html).

Any chance you could confirm the sdk version you are using? Do other calls succeed when the using the sdk?

**Comment by ranoichman** [comment]
Hi,

Looked again in all the files I see (Readme, etc.) and I see an .snk file
that I couldn't get a version of.. How can I find the version?

Regarding the rest of the calls - all works, this call also works when the
configuration is correct (i.e permissions, token), just when something is
not right, instead of a quick failure like in most other cases, I get hang
and thrown like I mentioned.

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by andrueastman** [comment]
I believe you should ideally have a `.csproj` file for your project that will contain a line like this which will show you the version. 

`<PackageReference Include="Microsoft.Graph" Version="x.x.x" />`

Any chance this is the case?

**Comment by ranoichman** [comment]
Hi,

Yes, of course this is the usual case, but unfortunately ours is a bit
different :
We don't consume the Nuget as is, but a few years ago we downloaded the
whole Sdk nuget, sort of a "snapshot" of the code and use that (the goal
was to have a more stable product as changes made to the sdk by Ms would
require us patches and risk our product).

In the library of the code, is there a specific file that states the
version?

**Comment by andrueastman** [comment]
Thanks for the extra information @ranoichman.

You should find a `Microsoft.Graph.csproj` in the folder `src/Microsoft.Graph/`. The version should be set in the `VersionPrefix` tag.

**Comment by ranoichman** [comment]
Hi,

Sorry for the hard time, but I went over the csproj and there is no version
mentioned there.... We are using the Beta version, I don't believe it's
different there, right?

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by andrueastman** [comment]
Yeah. You should be able to find the version in a location similar to the link below.

https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/blob/3f2901015179a92004fc08fb8e474d28df56659c/src/Microsoft.Graph/Microsoft.Graph.Beta.csproj#L28

**Comment by ranoichman** [comment]
Hi,

It's more complicated than that, I'm afraid. It seems that we copied the
source code to our own structure and for the life of me I can't find the
version anywhere... is it critical? I know it to be a version from mid
2018, about April I think, but that's it...

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by andrueastman** [comment]
Are you able to extract the version of the `Microsoft.Graph.Core` used as dependency in the csproj file?
It might be worth finding that out as the core package does the work of sending out the requests and httpClient configuration.

**Comment by ranoichman** [comment]
It's the same with it as well - was copied and wrapped so I can't see the version, but as I mentioned - the calls work, it's just specific scenarios like I described that make the calls "hang" for no obvious reason

**Comment by ranoichman** [comment]
Well I dug a bit deeper, and I found it - we use version 1.12!

**Comment by andrueastman** [comment]
Would you be in a position to at least upgrade the code of `Microsoft.Graph.Core` to the latest stable 1.x.x version and confirm if the issue still persists?

Alot has changed since version 1.12 and I suspect the issue could have already been resolved. 

As we have moved up to version 2 with several breaking changes, you will find the code of the latest stable 1.x.x version at the head of `support/1.x` branch [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/tree/support/1.x).

**Comment by ranoichman** [comment]
I'm afraid of the implications in this kind of move. Is it possible to only
upgrade the Core and leave the Beta at the same version it is now?

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by andrueastman** [comment]
Yes, it should be possible as there should be no breaking changes between core 1.12 and the latest 1.x.x core version at the head of the branch I provided above.
